src/explore-education-statistics-admin/src/pages/methodology/edit-methodology/content/context/MethodologyContext.tsx

```diff
@@ -0,0 +1,176 @@
+import remove from 'lodash/remove';
+import React, { createContext, ReactNode, useContext } from 'react';
+import { MethodologyContent } from 'src/services/methodology/types';
+import { Reducer, useImmerReducer } from 'use-immer';
+import MethodologyDispatchAction from './MethodologyContextActionTypes';
+
+type MethodologyContextDispatch = (action: MethodologyDispatchAction) => void;
+type MethodologyContextState = {
+  methodology: MethodologyContent | undefined;
+  canUpdateMethodology: boolean;
+};
+type MethodologyProviderProps = { children: ReactNode };
+
+const MethodologyStateContext = createContext<
+  MethodologyContextState | undefined
+>(undefined);
+const MethodologyDispatchContext = createContext<
+  MethodologyContextDispatch | undefined
+>(undefined);
+
+export const methodologyReducer: Reducer<
+  MethodologyContextState,
+  MethodologyDispatchAction
+> = (draft, action) => {
+  switch (action.type) {
+    case 'CLEAR_STATE': {
+      return {
+        methodology: undefined,
+        canUpdateMethodology: false,
+      };
+    }
+    case 'SET_STATE': {
+      return { ...draft, ...action.payload };
+    }
+    case 'REMOVE_BLOCK_FROM_SECTION': {
+      const { sectionId, blockId, sectionKey } = action.payload.meta;
+      if (!draft.methodology?.[sectionKey]) {
+        throw new Error('REMOVE_BLOCK_FROM_SECTION: failed');
```

Comment:
Let's provide a more detailed error message like 'REMOVE_BLOCK_FROM_SECTION: Section could not be found`. The same should be done for the other failures